sdk/core/azure-core/azure/core/pipeline/policies/_utils.py

```diff
@@ -102,3 +102,28 @@ def get_domain(url: str) -> str:
     :return: The domain of the url.
     """
     return str(urlparse(url).netloc).lower()
+
+
+def sanitize_url_query_params(url: str, allowed_query_params: Set[str], placeholder: str = "REDACTED") -> str:
+    """Redact query parameters from the specified URL.
+
+    :param url: The URL to sanitize.
+    :type url: str
+    :param allowed_query_params: The set of query parameters to allow.
+    :type allowed_query_params: set[str]
+    :param placeholder: The placeholder to use for redacted query parameters.
+    :type placeholder: str
+    :rtype: str
+    :return: The sanitized URL.
+    """
+    allowed = allowed_query_params
+    # In the majority of cases, the allowed query parameters will be a CaseInsensitiveSet. If it's not,
```

Comment:
When is this not a CaseInsensitiveSet? And why do we care?

Reply:
Found a singular case where an SDK made a custom `HttpLoggingPolicy` that set the header allow list like so:
```python
self.allowed_header_names = set(cosmos_allow_list)
```

Not a common occurrence as most SDKs just use `allowed_header_names.update` to make changes, but I still want to to maintain the case-insensitive redaction if someone does this with `allowed_query_params`.

Reply:
I know for headers we are trying to change the casing of `Bearer` in generated tsp code, so casing does matter in that context. Not sure how much it relates to queries though, but pasting here as an fyi https://github.com/microsoft/typespec/issues/2672